tests/unit/fem/test_assemblediagonalpa.cpp

```diff
@@ -209,10 +208,8 @@ TEST_CASE("diffusiondiag")
                }
                else if (coeffType == 3)
                {
-                  mcoeff = new MatrixFunctionCoefficient(dimension,
```

Comment:
It looks like you removed the testing of `fullSymmetricMatrixCoeffFunction`, which is not really necessary. In that case, the function `fullSymmetricMatrixCoeffFunction` could be removed.

Reply:
Good point, yes they are not used now because the legacy integrator can directly use the symmetric coefficients.

I removed those functions now.